### ISSUE TITLE
Add configuration with binding at runtime

### ISSUE THREAD

**Issue by MauriceNino** [JonasAlfredsson/docker-nginx-certbot#33]
With the current setup it is only possible to add custom configuration files with a custom Dockerfile:

```
FROM jonasal/nginx-certbot:latest
COPY conf.d/* /etc/nginx/conf.d/
```

but it would be easier to use, when you could do something like this:

```
docker container run -itd \
        -v /host/nginx.conf:/etc/nginx/nginx.conf:ro \
        -v /host/nginx_secrets:/etc/letsencrypt \
        -v /host/conf.d:/etc/nginx/conf.d/ \ # <-- problem here
        -e CERTBOT_EMAIL=mauriceprivat98@gmail.com \
        jonasal/nginx-certbot
```
 
The problem here is the line: `-v /host/conf.d:/etc/nginx/conf.d/`, because it will overwrite what is inside the `conf.d` folder.

A possible solution would be:

1. Mount the configuration's folder to something like `-v /host/conf.d:/etc/nginx_conf:ro` (note: readonly)
2. At runtime, before doing anything else, copy the content to the real configuration's folder: `cp /etc/nginx_conf/* /etc/nginx/conf.d/`

**Comment by JonasAlfredsson** [comment]
I have thought about this, and up until now I have liked the current approach as it allows you (as a user) to very easily just build additional stuff into the container or manually copy the files from this repo into your host mounted folder to really see everything that is loaded when starting this container. 

My reasoning behind this is that it should be as easy as possible to overwrite/remove my configuration files if you do not want to have my "redirector.conf" for example. I also think that having a "copy" step at startup will add more complexity in regards to file checking and live reloading of configuration files. So unless this is a more widely requested feature I don't think I will implement it into this project, as it is also not how the previous staticfloat image worked. 

Hope this doesn't cause too much trouble for you :)
//Jonas

**Comment by MauriceNino** [comment]
Yeah, I think it is an OK solution as well, just thought it is kind of an inconvenience, because you can't really use the image without writing your own Dockerfile. Of course, if you have more specific tasks or modifications to do, you are more or less bound to extend the image, but for simple use-cases, it should be possible to directly use it from DockerHub. 

I kind of fail to understand, why the copy step at startup adds more complexity, while renaming files in the live directory at runtime is fine?

Ultimately it is your repository and therefore your decision, so I am fine - the PR is to be seen as a possible solution. For me, it doesn't change much, I just thought it would be a really convenient feature. ^^

PS: There is another solution that works with the current image, but you have to bind the local conf.d folder to the containers' conf.d folder, which overwrites your configs AND has to be done in rw mode, because otherwise the `mv ... ...` fails.

**Comment by JonasAlfredsson** [comment]
With more complexity I mean we should probably have some checks so we do not overwrite potential user configurations, while at the same time do overwrite "my" configuration files that may have been updated here in the repository. The last part is sort of already a problem today, so perhaps we can put the current files in some other directory and symlink to them....

This deserves some more thinking, so I will take some time during the weekend, but I am still skeptical.

**Comment by JonasAlfredsson** [comment]
I have been working a bit on this, and I think I have found a way that would allow you to just host mount directy linto the `/etc/nginx/user_conf.d/`, and have those files be symlinked into the "normal" `/etc/nginx/conf.d/` folder. 

I will build and push this to the dev branch so you can test this out if you want later. You _should_ only have to do this then:

```
docker run -it \
        -v /host/nginx.conf:/etc/nginx/nginx.conf:ro \
        -v /host/nginx_secrets:/etc/letsencrypt \
        -v /host/conf.d:/etc/nginx/user_conf.d:ro
        -e CERTBOT_EMAIL=somemail@example.com \
        jonasal/nginx-certbot:dev
```

**Comment by JonasAlfredsson** [comment]
Excellent questions, I will see if I can answer them to your satisfaction :)

1. I placed it inside the loop since this is loop is re-triggered when you send in a HUP signal => i.e. when you tell Nginx to reload changes. So even if the changes/symlinks were added immediately, Nginx would not pick them up until you tell it to reload. And even then we need to make sure that all the files/certificates are present or else it will crash. So the way to do it would be to change files in your host mounted folders and then do a `docker kill --signal=HUP nginx-certbot` to have changes be propagated. 

(1.5. Unless the config file is removed during the millisecond it takes to find all file paths necessary for it, I think we will handle it very well, since the other programs running in the loop will handle files that do not have all their certs yet.)

2. The is a good and valid question, and this extra check is only done because it was dead simple to add, and will give the user a hint of what can have gone wrong (e.g. the user might host mount both of these folders and make their own symlinks between them).

**Comment by MauriceNino** [comment]
Okay I understand where you are coming from! I would make this very clear in the documentation though, to avoid any confusion.

For the extra step: I just thought it complicates stuff a bit, but I am nowhere near a good bash scripter, so I might be the wrong person to ask anyways :P

Btw, I just built & ran the dev branch on an Ubuntu Server, and it works flawlessly as far as I can see :)

**Comment by JonasAlfredsson** [comment]
Great, I will continue to tinker with this a little bit more, but this will most likely be a 2.0.0 release since this is such a large change. 
I will also make sure that the documentation reflects all the changes when I am happy with how it works.

**Comment by MauriceNino** [comment]
Cool! Feel free to @ me if I can help you out in any way :)

**Comment by JonasAlfredsson** [comment]
I think I am happy with how it turned out. It is now released as v2.0.0 :)

**Comment by JonasAlfredsson** [comment]
Furthermore, if you have any comments/suggestions in regards to the documentation I am all ears :)
